certbot-apache/certbot_apache/_internal/dualparser.py

```diff
@@ -202,9 +214,8 @@ def add_child_block(
         assertions.assertEqual(primary_new, secondary_new)
         return DualBlockNode(primary=primary_new, secondary=secondary_new)
 
-    def add_child_directive(
-        self, name: str, parameters: Optional[str] = None, position: Optional[int] = None
-    ) -> DualDirectiveNode:
+    def add_child_directive(self, name: str, parameters: Optional[str] = None,
+                            position: Optional[int] = None) -> DualDirectiveNode:
```

Comment:
I am a little perplexed as to why this passes type checking. `primary` and `secondary` take `Optional[Sequence[str]]` and `Optional[str]` for `parameters`, respectively 😕 .

`ApacheBlockNode.add_child_directive` doesn't match `interfaces.BlockNode.add_child_directive` either, though it seems like it should.

Reply:
Well, I think it is because the `Sequence` contract for strings is quite unfortunate for type checking. Indeed, a `str` comply to `Sequence[str]`. It is sizeable (`len("test") == 4`) and you can access to values by numeric indexes while these values are `str` (`"test"[0] == "t"`).

So I prefer to not follow the "most generic type possible" rule here to have something more significant and require `List[str]` in the contracts. I also removed the other `Sequence[str]` in the module to avoid confusions.